nevergrad/optimization/optimizerlib.py

```diff
@@ -256,6 +257,82 @@ def __init__(
 CMA = ParametrizedCMA().set_name("CMA", register=True)
 DiagonalCMA = ParametrizedCMA(diagonal=True).set_name("DiagonalCMA", register=True)
 
+class _FCMA(base.Optimizer):
+
+    def __init__(
+            self,
+            parametrization: IntOrParameter,
+            budget: Optional[int] = None,
+            num_workers: int = 1,
+            scale: float = 1.0,
+            popsize: int = 0
+    ) -> None:
+        super().__init__(parametrization, budget=budget, num_workers=num_workers)
+        self._scale = scale
+        self._popsize = popsize if popsize > 0 else \
+                            max(self.num_workers, 4 + int(3 * np.log(self.dimension)))
+        self._es: Optional[cmaes.Cmaes] = None
+        # delay initialization to ease implementation of variants
+        self.listx: tp.List[ArrayLike] = []
+        self.listy: tp.List[float] = []
+        self.to_be_asked: tp.Deque[np.ndarray] = deque()
+
+    @property
+    def es(self) -> cmaes.Cmaes:
+        if self._es is None:
+            self._es = cmaes.Cmaes(x0=np.zeros(self.dimension, dtype=np.float),
+                                   input_sigma=self._scale,
+                                   popsize=self._popsize, randn=self._rng.randn)
+        return self._es
+
+    def _internal_ask(self) -> ArrayLike:
+        if not self.to_be_asked:
+            self.to_be_asked.extend(self.es.ask())
+        return self.to_be_asked.popleft()
+
+    def _internal_tell(self, x: ArrayLike, value: float) -> None:
+        self.listx += [x]
+        self.listy += [value]
+        if len(self.listx) >= self.es.popsize:
+            try:
+                self.es.tell(self.listy, self.listx)
+            except RuntimeError:
+                pass
+            else:
+                self.listx = []
+                self.listy = []
+
+    def _internal_provide_recommendation(self) -> ArrayLike:
+        if self._es is None:
+            raise RuntimeError("Either ask or tell method should have been called before")
+        if self.es.best_x is None:
+            return self.current_bests["pessimistic"].x
+        return self.es.best_x  # type: ignore
+
+
+class ParametrizedFCMA(base.ConfiguredOptimizer):
+    """FCMA-ES optimizer, wrapping external implementation: https://github.com/dietmarwo/fast-cma-es
```

Comment:
You should add a few more lines about how different it is from the original implementation and in which case we should use your implementation

Reply:
Ok. Short answer: If you don't use DiagonalOnly FCMA should be the better/faster option. At least this is what my experiments using  your benchmarks suggest. But with both variants
it is really important to use "export MKL_NUM_THREADS=1" or "export OPENBLAS_NUM_THREADS=1" for OPENBLAS if num_workers > 1 for the benchmark.
When BLAS multithreading conflicts with multithreading at the benchmark level CMA can slow down up to factor 10 for higher dimensions.

Reply:
Ok. Short answer: If you don't use DiagonalOnly FCMA should be the better/faster option. At least this is what my experiments using your benchmarks suggest. But with both variants
it is really important to use "export MKL_NUM_THREADS=1" or "export OPENBLAS_NUM_THREADS=1" for OPENBLAS if num_workers > 1 for the benchmark.
When BLAS multithreading conflicts with multithreading at the benchmark level CMA can slow down up to factor 10 for higher dimensions.

Reply:
Maybe we need to add a context manager which sets this variables when calling CMA ? Inside ask and or tell in order to avoid border effects

Reply:
fcmaes provides a retry mechanism where I set these variables internally. Tried that with Nevergrad and it failed somehow, probably because they were not propagated to the workers. Will investigate this further.

Reply:
Can you elaborate on the mechanism?